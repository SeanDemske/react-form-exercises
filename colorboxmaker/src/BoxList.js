import React, { useState } from "react";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";
import { v4 as uuid } from 'uuid';

const BoxList = () => {
    const [boxes, setBoxes] = useState([]);

    const addBox = (boxData) => {
        setBoxes([...boxes, {
            ...boxData,
            id: uuid()
        }]);
    }

    const removeBox = (id) => {
        setBoxes(boxes.filter(box => box.id !== id))
    }

    return (
        <div className="BoxList">
            <NewBoxForm addBox={addBox} />
            <h1>BoxList</h1>
            {boxes.map(box => (
                <Box 
                    id={box.id}
                    width={box.width}
                    height={box.height}
                    color={box.color}
                    removeBox={removeBox}
                />
            ))}
        </div>
    )
}

export default BoxList;