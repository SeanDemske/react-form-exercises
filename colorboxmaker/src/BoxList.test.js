import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import BoxList from "./BoxList";

function addBox(boxList, height = "2", width = "2", color = "peachpuff") {
  const heightInput = boxList.getByLabelText("Height:");
  const widthInput = boxList.getByLabelText("Width:");
  const backgroundInput = boxList.getByLabelText("Color:");
  fireEvent.change(backgroundInput, { target: { value: color } });
  fireEvent.change(widthInput, { target: { value: width } });
  fireEvent.change(heightInput, { target: { value: height } });
  const button = boxList.getByText("Submit");
  fireEvent.click(button);
}

it("renders without crashing", function() {
  render(<BoxList />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

// it("can add a new box", function() {
//   const boxList = render(<BoxList />);

//   // no boxes yet
//   expect(boxList.queryByText("Delete")).not.toBeInTheDocument();

//   addBox(boxList);

// });
