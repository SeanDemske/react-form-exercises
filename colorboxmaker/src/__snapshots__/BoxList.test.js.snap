// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form
      class="NewBoxForm"
    >
      <label
        for="width-input"
      >
        Width:
      </label>
      <input
        id="width-input"
        name="width"
        type="text"
        value=""
      />
      <label
        for="height-input"
      >
        Height:
      </label>
      <input
        id="height-input"
        name="height"
        type="text"
        value=""
      />
      <label
        for="color-input"
      >
        Color:
      </label>
      <input
        id="color-input"
        name="color"
        type="text"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <h1>
      BoxList
    </h1>
  </div>
</DocumentFragment>
`;
